docs/api/utility-process.md

```diff
@@ -0,0 +1,128 @@
+# UtilityProcess
+
+`UtilityProcess` creates a child process with
+Node.js and Message ports enabled. It provides the equivalent of [`child_process.fork`][] API from Node.js
+but instead uses [Services API][] from Chromium to launch the child process.
+
+`UtilityProcess` is an [EventEmitter][event-emitter].
+
+## Class: UtilityProcess
+
+> Child process with Node.js integration spawned by Chromium.
+
+Process: [Main](../glossary.md#main-process)<br />
+
+### `new UtilityProcess(modulePath[, args][, options])`
```

Comment:
is `new` the right way to express this? Should this be `utilityProcess.spawn()` or `process.spawnUtility()` or something else? I find it a little awkward that `new` has the side effect of spawning a process; that feels like it should be a function call. Somewhat weak opinion, though, I'm willing to be convinced otherwise.

Reply:
I wanted the call to represent that users have to work with an instance of the `UtilityProcess` module for every process, mostly inspired from `BrowserWindow`. But I could also wrap this in a function call which would internally just perform the same `new` call, I think an example of this would be `net.request`. Also, the module itself don't have the necessity for any static methods, all options that could modify the process behavior have to be done before the process is launched and any interaction after that will be with the process instance, so I went with the `new` interface.

Reply:
Since this API is inspired by `child_process.fork()` (which does return a `ChildProcess` object), I think it makes sense for this to be a regular function rather than a constructor. I suppose our `EventEmitter` machinery requires `gin::Constructible` anyway, but that's an implementation detail as you illustrate.

I put in a weak vote for `utilityProcess.spawn()`, maybe @MarshallOfSound could weigh in and break the tie?

Reply:
Whether to use constructor or factory function depends on how the APIs might be extended in future.

While this API is only about executing Node.js scripts, I think it is possible that users may want APIs to execute shell commands directly, i.e. `utilityProcess.execFile`, `utilityProcess.exec` and `utilityProcess.spwan`. The PR body also mentioned possible deno support, so maybe an `utilityProcess.forkDeno` API in future.

(I prefer `utilityProcess.fork` over `utilityProcess.spawn` for this API, because `fork` execute Node.js script, while `spawn` executes shell command).

Reply:
We have a tie breaker, I will rewire this to be `utilityProcess.fork`

Reply:
Refs https://github.com/electron/electron/pull/34980/commits/6616a24453464428b75710ac8116e0e5cc5152f8